### PR TITLE
Add documentation for setting proxy

### PR DESCRIPTION
**Pull request by Stiv-work**
Document setting proxy

- [x]  Check developer mode link is valid, before merging in.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -391,6 +391,35 @@ For more examples and ideas, visit:
 
 ```
 
+#### Proxy Setup
```
Comment (madelen-at-work) [inline README.md:394]: I think you should add a few lines on why someone would need to set the proxy env variables. e.g.
If the device is located behind a corporate proxy you may need to set the correct environment variables. This is done by configuring proxy behavior for dockerd in the daemon.json file as described in [Configure the Docker daemon to use a proxy server](https://docs.docker.com/config/daemon/systemd/#httphttps-proxy)][https://docs.docker.com/config/daemon/systemd/#httphttps-proxy].

To do this for the application you can either use [developer mode](#developermode) or build the application manually with a pre-creaeted daemon.json that is copied onto the device by the postinstall script 
^ The last section should be extended to be a bit more detailed in both cases

I do not think you should add the shell script - only describe what the daemon.json should contain, where it is located on the device and that the application needs to be restarted once it is updated.

File: README.md [hunk 2]
```diff
@@ -391,6 +391,35 @@ For more examples and ideas, visit:
...
+
+> [!NOTE]
+> **From AXIS OS 12.0**
+>
+> To use the shell script provided to set proxy, [developer mode](#developermode) certificate has to be set on camera, to be able use acap-dockerdwrapper ssh user instead of root ssh user.
```
Comment (killenheladagen) [inline README.md:399]: SSH as root requires another (Axis internal) certificate, so rephrase it something like: "... developer mode certificate has to be installed on the camera (device? which term is used), in order to use SSH."

Will the script take care of which username to use? I suggest that it defaults to acap-dockerdwrapper and only uses root if a special option "--root" or similar is passed in.

File: README.md (hunk 1 above)
Comment (Stiv-work) [reply README.md:394]: Removed the script and the note

File: README.md [hunk 3]
```diff
@@ -391,6 +391,24 @@ For more examples and ideas, visit:
...
+
+If the device is located behind a corporate proxy you may need to set the correct environment variables. This is done by configuring proxy behavior for dockerd in the daemon.json file as described in ['Configure the Docker daemon to use a proxy server'][docker-proxy].
+
+To do this for the application you can either use [developer mode][developermode] or build the application manually with a pre-created daemon.json that is copied onto the device by the postinstall script.
+daemon.json file should be located at /usr/local/packages/dockerdwrapper/localdata/daemon.json on the device and should include the fallowing properties.
+
+```json
+{
+  "proxies": {
+    "http-proxy": "http://<myproxy>:<port>",
```
Comment (madelen-at-work) [inline README.md:404]: use the same example here as on the docs.docker.com page

File: README.md [hunk 4]
```diff
@@ -391,6 +391,24 @@ For more examples and ideas, visit:
 
 ```
 
+#### Proxy Setup
+
+If the device is located behind a corporate proxy you may need to set the correct environment variables. This is done by configuring proxy behavior for dockerd in the daemon.json file as described in ['Configure the Docker daemon to use a proxy server'][docker-proxy].
+
+To do this for the application you can either use [developer mode][developermode] or build the application manually with a pre-created daemon.json that is copied onto the device by the postinstall script.
+daemon.json file should be located at /usr/local/packages/dockerdwrapper/localdata/daemon.json on the device and should include the fallowing properties.
```
Comment (madelen-at-work) [inline README.md:399]:
```suggestion
The daemon.json file should be located at `/usr/local/packages/dockerdwrapper/localdata/daemon.json` on the device and should include the following properties.
```

File: README.md [hunk 5]
```diff
@@ -391,6 +391,24 @@ For more examples and ideas, visit:
...
+```json
+{
+  "proxies": {
+    "http-proxy": "http://<myproxy>:<port>",
+    "https-proxy": "http://<myproxy>:<port>",
+    "no-proxy": "localhost,127.0.0.0/8,10.0.0.0/8,192.168.0.0/16,172.16.0.0/12,.<domain>"
+ }
+}
+```
+If [developer mode][developermode] is used, after downloading the application SSH user acap-dockerdwrapper will be created. Update the password of acap-dockerdwrapper. SSH user acap-dockerdwrapper can be used to write in the daemon.json file, application has to be restarted once daemon.json is updated.
```
Comment (madelen-at-work) [inline README.md:410]: Setting the contents of the daemon.json file can be done either by adding it to the source code and rebuilding the application or by ssh:ing into the device with an already installed application. In the latter case [developer mode][developermode] is needed, see that documentation for further details. Also note that, if the application is running when the file is updated, it needs to be restarted for the change to take effect.

**Review by madelen-at-work — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -407,7 +407,7 @@ daemon.json file should be located at /usr/local/packages/dockerdwrapper/localda
  }
 }
 ```
-If developer mode is used, after downloading the application, ssh user will be created. Update the password of the ssh user. Then the application ssh user can be used to write in the daemon.json file, application has to be restarted once updated
+If [developer mode][developermode] is used, after downloading the application SSH user acap-dockerdwrapper will be created. Update the password of acap-dockerdwrapper. SSH user acap-dockerdwrapper can be used to write in the daemon.json file, application has to be restarted once daemon.json is updated.
```
Comment (killenheladagen) [inline README.md]:
```suggestion
If [developer mode][developermode] is used, the SSH user acap-dockerdwrapper will be created when the application is installed. Once the password of acap-dockerdwrapper has been set, SSH can be used to edit daemon.json file. The application has to be restarted in order for changes in daemon.json to take effect.
```

Comment (Stiv-work) [reply README.md]: This part has been changes

**Review by madelen-at-work — COMMENTED** [review]
I think the text is ok now but we need to wait for the Developer Mode section in the documentation

File: README.md [hunk 7]
```diff
@@ -391,6 +392,32 @@ For more examples and ideas, visit:
...
+  "proxies": {
+    "http-proxy": "http://proxy.example.com:3128",
+    "https-proxy": "https://proxy.example.com:3129",
+    "no-proxy": "*.test.example.com,.example.org,127.0.0.0/8"
+  }
+}
+```
+
+Setting the contents of the daemon.json file can be done either by adding it to the source code and
+rebuilding the application or by ssh:ing into the device with an already installed application and updating
```
Comment (pataxis) [inline README.md:415]: I think this sounds a bit better than "ssh:ing"
Reference: https://www.redhat.com/sysadmin/access-remote-systems-ssh 

```suggestion
rebuilding the application or by logging into the device over SSH with an already installed application and updating
```

File: README.md [hunk 8]
```diff
@@ -391,6 +392,32 @@ For more examples and ideas, visit:
...
+the file.
+In the latter case [developer mode][developermode] is needed, see that documentation for further details.
```
Comment (pataxis) [inline README.md:417]: In acap-documentation we have chosen to use Developer Mode over developer mode.
```suggestion
In the latter case [Developer Mode][developermode] is needed, see that documentation for further details.
```

File: README.md [hunk 9]
```diff
@@ -391,6 +392,32 @@ For more examples and ideas, visit:
 
 ```
 
+#### Proxy Setup
+
+If the device is located behind a corporate proxy you may need to set the correct environment variables.
```
Comment (pataxis) [inline README.md:397]: In the Docker documentation there are two ways to set a proxy,

- configuring the daemon
- setting environment variables

These instructions configures the daemon, so it seems a bit strange to mention environment variables.

Maybe something like this?
```suggestion
If the device is located behind a proxy the Docker daemon needs to be configured.
```

**Review by madelen-at-work — APPROVED** [review]

**Pull request merged**